File: THCS/2021_2022/HCM_2122/Bai1/SODOCLAP.cpp
#include <iostream>
#include <fstream>
#include <set>
#include <string>
using namespace std;

// Hàm kiểm tra một số có phải số độc lập hay không
bool isIndependent(int n) {
    set<char> digits;
    string s = to_string(n);
    for (char c : s) {
        if (digits.find(c) != digits.end()) return false;
        digits.insert(c);
    }
    return true;
}

int main() {
    // Đọc dữ liệu từ file SODOCLAP.INP
    ifstream inputFile("SODOCLAP.INP");
    ofstream outputFile("SODOCLAP.OUT");

    int X;
    inputFile >> X;

    // Tìm số độc lập nhỏ nhất lớn hơn X
    int result = X + 1;
    while (!isIndependent(result)) {
        result++;
    }

    // Ghi kết quả ra file SODOCLAP.OUT
    outputFile << result << endl;

    inputFile.close();
    outputFile.close();

    return 0;
}

File: THCS/2021_2022/HCM_2122/Bai1/SODOCLAP.py
def is_independent(n):
    # Kiểm tra xem các chữ số trong n có xuất hiện trùng lặp hay không
    digits = str(n)
    return len(set(digits)) == len(digits)

# Đọc dữ liệu từ file SODOCLAP.INP
with open("SODOCLAP.INP", "r") as file_in:
    X = int(file_in.read().strip())

# Tìm số độc lập nhỏ nhất lớn hơn X
result = X + 1
while not is_independent(result):
    result += 1

# Ghi kết quả ra file SODOCLAP.OUT
with open("SODOCLAP.OUT", "w") as file_out:
    file_out.write(str(result) + "\n")
